TEC-4140: Coupon related components

joelsalmeida opened the pull request.

joelsalmeida committed:
TEC-4140: creates "CollapsibleContent" component

joelsalmeida committed:
TEC-4140: creates "UnstyledButton" component

joelsalmeida committed:
TEC-4140: add some stories

gabrielcdiniz commented on src/components/Collapsible/CollapsibleContent/style.tsx:
```diff
@@ -0,0 +1,11 @@
+import styled from 'styled-components'
+import { Colors } from '../../../tokens'
+
+const StyledCollapsibleContent = styled.div<{ $open?: boolean }>`
```
rename prop `$open` to `open`. Do not need to add dolar sign

gabrielcdiniz replied:
rename "styles", to contain suffix `Styled` at final. E.g `CollapsibleContentStyled`

gabrielcdiniz commented on src/components/UnstyledButton/index.tsx:
```diff
@@ -0,0 +1,15 @@
+  ...props
+}: UnstyledButtonComponentProps) => {
+  return <StyledUnstyledButton {...props}>{children}</StyledUnstyledButton>
+}
```
this component must be inside "Button" folder, because he is a variant of button, not a new component. And your name must be "Ghost Button".

Here comes an example of "Ghost Button":
- https://carbondesignsystem.com/components/button/style/#ghost-button

gabrielcdiniz commented on src/components/UnstyledButton/styles.tsx:
```diff
@@ -0,0 +1,22 @@
+  &:disabled {
+    cursor: not-allowed;
+    color: ${Colors.light.neutral.neutral200};
+  }
```
do not need this, because "disabled" will be a button variant

gabrielcdiniz commented on src/stories/Collapsible.stories.tsx:
```diff
@@ -0,0 +1,78 @@
+    <div>
+      <StyledCollapsibleContent $open={open}>
+        {content}
+      </StyledCollapsibleContent>
```
here always be the component, never the "styled" component. E.g: 

```tsx
<CollapsibleContent open={open}>
  {content}
</CollapsibleContent>
```

gabrielcdiniz commented on src/stories/Collapsible.stories.tsx:
```diff
@@ -0,0 +1,78 @@
+  )
+}
+
+export default {
+  title: 'Components/Collapsible',
+  component: CollapsibleToggleComponent,
+  decorators: [
+    (Story) => (
+      <StoryPlaceholder>
+        <StyledSpan>
+          You can use a helper function with a button to toggle the component.
+        </StyledSpan>
+
+        <Story />
+      </StoryPlaceholder>
+    ),
+  ],
+} as Meta
```
for convenience, meta must be always on top file and follows the pattern:

![image](https://github.com/vissimo-group/chardonnay/assets/53409573/d9c2c20a-58fc-463d-bf28-09550d10cffb)

the print has taken from storybook docs website

gabrielcdiniz commented:
@joelsalmeida take some printscreens for visual validations

joelsalmeida committed:
TEC-4140: change to composition pattern

joelsalmeida committed:
TEC-4140: adjust imports and exports

joelsalmeida committed:
TEC-4140: add context

joelsalmeida committed:
TEC-4140: adjust stories

joelsalmeida committed:
TEC-4140: Refactor CollapsibleTrigger component to use styled button

This commit refactors the CollapsibleTrigger component to use a styled button instead of a regular button. It also adds support for passing children and an onToggle function as props.

joelsalmeida committed:
TEC-4140: Add transition animation to CollapsibleContent

joelsalmeida committed:
TEC-4140: Refactor CollapsibleContext to use useState and useEffect

joelsalmeida committed:
TEC-4140: Refactor CollapsibleRoot component to use CollapsibleProvider

joelsalmeida committed:
TEC-4140: Add @storybook/preview-api dependency and update Collapsible.stories.tsx

joelsalmeida committed:
TEC-4140: Delete UnstyledButton component and related files

gabrielcdiniz commented on src/components/Collapsible/CollapsibleTrigger/index.tsx:
```diff
@@ -0,0 +1,24 @@
+    toggleIsCollapsed()
+    onToggle?.()
+  }
+
```
use `useCallback` hook wrapping functions for performance

gabrielcdiniz commented on src/components/Collapsible/CollapsibleContent/index.tsx:
```diff
@@ -0,0 +1,19 @@
+  const { isCollapsed } = useContext(CollapsibleContext)
+
+  return (
+    <StyledCollapsibleContent $open={isCollapsed} {...props}>
```
rename prop `$open` to `open`. Do not need to add dolar sign

joelsalmeida replied:
In this case it is necessary because it is a styled components prop.
If I try to remove the "$" it triggers a type error.

joelsalmeida committed:
TEC-4140: Rename "StyledCollapsibleContent" component to "CollapsibleContentStyled"

joelsalmeida committed:
TEC-4140: Refactor CollapsibleTrigger component to use useCallback hook

joelsalmeida committed:
TEC-4140: Adjust Collapsible.stories

samuelvermeuln commented on src/stories/Collapsible.stories.tsx:
```diff
@@ -0,0 +1,116 @@
+  width: 600px;
+  padding: ${Spacing['5']};
+  border-radius: ${BorderRadius['3']};
+  border: 1px solid ${Colors.light.neutral.neutral200};
```
the theme must accept dark mode

joelsalmeida replied:
This is only used for presentation, it isn't part of the component itself.

samuelvermeuln commented on src/components/Collapsible/CollapsibleContent/style.tsx:
```diff
@@ -0,0 +1,12 @@
+  height: ${({ $open }) => ($open ? 'fit-content' : '0px')};
+  opacity: ${({ $open }) => ($open ? '1' : '0')};
+  overflow: hidden;
+  color: ${Colors.light.neutral.neutral400};
```
the theme must accept dark mode

samuelvermeuln commented on src/stories/Collapsible.stories.tsx:
```diff
@@ -0,0 +1,116 @@
+  border: 1px solid ${Colors.light.neutral.neutral200};
+`
+const StyledSpan = styled.header`
+  color: ${Colors.light.neutral.neutral400};
```
the theme must accept dark mode

joelsalmeida replied:
This is only used for presentation, it isn't part of the component itself.

joelsalmeida committed:
TEC-4140: Add React import to Collapsible components

joelsalmeida committed:
TEC-4140: Add theme prop to CollapsibleContent component

joelsalmeida committed:
4140: Export Collapsible component in index.ts

joelsalmeida committed:
Merge branch 'main' into TEC-4140/feature/coupon-related-components

joelsalmeida committed:
TEC-4140: Fix import/export of TagCep and InfoTopBar
